mangoPURE/data/transforms.py

```diff
@@ -0,0 +1,280 @@
+            speaker_audio = speaker_audio[:audio_len]
+            speaker_len = audio_len
+
+        pos_start = random.randint(0, audio_len - 1)
+        if pos_start + speaker_len <= audio_len:
+            example.audio[pos_start:pos_start + speaker_len] += speaker_audio
+            start, length = pos_start, speaker_len
+        else:
+            end_len = audio_len - pos_start
+            start_len = speaker_len - end_len
+            if start_len > end_len:
+                example.audio[:start_len] += speaker_audio[-start_len:]
+                start, length = 0, start_len
+            else:
+                example.audio[-end_len:] += speaker_audio[:end_len]
+                start, length = pos_start, end_len
+
+        if example.noises_info is None:
+            example.noises_info = []
+        example.noises_info.append(SegmentInfo(
+            start=start,
+            length=length,
+            class_id=speaker_example["speaker_id"]
+        ))
+
+        return example
+
+
+class AddSeveralRandomSpeakerSegments(AddRandomSpeakerSegment):
+    def __init__(
+            self,
+            speaker_dataset: HfDataset,
+            min_speakers: int = 2,
+            max_speakers: int = 4,
+    ):
+        super().__init__(speaker_dataset)
+        self.min_speakers = min_speakers
+        self.max_speakers = max_speakers
+
+    def __call__(
+            self,
+            example: MixedExample
+    ) -> MixedExample:
+        """
+        The same as AddRandomSpeakerSegment, but it generates
+        several (random number) speakers in the audio
+        """
+        speakers_num = random.randint(self.min_speakers, self.max_speakers)
+        for i in range(speakers_num):
+            example = super().__call__(example)
+        return example
+
+
+class AddRandomFilledNoise(MixerTransform):
+    def __init__(
+            self,
+            noise_dataset: HfDataset,
+            snrs: tuple[float] = (10.0, 15.0, 20.0),
+            no_noise_coef: bool = True
+    ):
+        """
+        :param noise_dataset: the dataset of noises audio
+            It has to havc "audio":"array" and "noise_id" columns
+        :param snrs: __call__ generates random noise coefficient from this list
+        :param no_noise_coef: if False does not use noise coefficient at all
+        """
+        self.noises_dataset = noise_dataset
+        self.snrs = snrs
+        self.no_noise_coef = no_noise_coef
+
+    def __call__(
+            self,
+            example: MixedExample,
+    ) -> MixedExample:
+        """
+        It fills the audio with random noise from dataset
+        """
+        random_pos = random.randint(0, len(self.noises_dataset) - 1)
+        noise_example = self.noises_dataset[random_pos]
+        noise_len = example.audio.shape[0]
+        noise_audio = expand_tensor(noise_example["audio"]["array"], noise_len)
+
+        noise_coef = calc_audio_adjustment_coef(example.audio, noise_audio, self.snrs, self.no_noise_coef)
```

Comment:
I am not sure what is better. Solution without crutch - store multiple audio tracks and add "AudioMergerTransform" - it will add them together, calculate correct snr etc. 

So internal example will be:
```
audio_tracks: List[(tensor, is_noise)]
```

And on the outside we will always call this merger that will merge tracks into one.